src/test/java/sorald/miner/WarningMinerTest.java

```diff
@@ -176,7 +175,7 @@ public void extractWarnings_throwsException_whenCheckCrashes() {
                 };
 
         assertThrows(
-                AnalysisException.class,
+                IllegalArgumentException.class,
```

Comment:
@slarse Can you explain why the exception type is changed? I don't get it.

Reply:
That does look a bit odd. It's probably a quick-fix I applied without thinking too much about it. I'll have a look

Reply:
It was testing entirely the wrong thing. With the use of the new higher-level components, I've not been able to figure out how to load and run a custom check. This isn't something that we actually have a need to do, and so I've simply removed the functionality that this test tested (i.e. that Sorald crashes if a Sonar check crashes).